File: contracts/animation_request.py
import time
from dataclasses import dataclass, field, asdict
from typing import Dict, Any, List

@dataclass
class AnimationRequest:
    """
    Standardized AnimationRequest Data Contract (v1.0.0).
    Contains requested animation category, clip identifier or procedural system identifier,
    target layers, blend weight, transition duration, priority, interruption policy, and source module.
    """
    version: str = "1.0.0"
    request_id: str = ""
    timestamp: float = field(default_factory=time.time)
    category: str = "idle"
    clip_or_procedural_id: str = ""
    target_layers: List[str] = field(default_factory=lambda: ["Base Layer"])
    blend_weight: float = 1.0
    transition_duration: float = 0.3
    priority: int = 0  # Higher value = higher priority
    interruption_policy: str = "interrupt_if_higher"  # "allow", "block", "interrupt_if_higher"
    source_module: str = "AnimationPlanner"
    parameters: Dict[str, Any] = field(default_factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        return asdict(self)

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> "AnimationRequest":
        if not data:
            return cls()
        known = {k: v for k, v in data.items() if k in cls.__dataclass_fields__}
        return cls(**known)

File: Mate-Engine/Assets/MATE ENGINE - Scripts/Contracts/AnimationRequest.cs
using System;
using System.Collections.Generic;
using UnityEngine;

namespace Vivy.Contracts
{
    /// <summary>
    /// Standardized AnimationRequest Data Contract (v1.0.0).
    /// Mirror of Python AnimationRequest dataclass.
    /// </summary>
    [Serializable]
    public class AnimationRequest
    {
        public string version = "1.0.0";
        public string request_id = "";
        public double timestamp = 0;
        public string category = "idle";
        public string clip_or_procedural_id = "";
        public List<string> target_layers = new List<string> { "Base Layer" };
        public float blend_weight = 1.0f;
        public float transition_duration = 0.3f;
        public int priority = 0;
        public string interruption_policy = "interrupt_if_higher";
        public string source_module = "AnimationPlanner";

        public static AnimationRequest FromJson(string json)
        {
            if (string.IsNullOrEmpty(json)) return new AnimationRequest();
            try { return JsonUtility.FromJson<AnimationRequest>(json); }
            catch (Exception ex)
            {
                Debug.LogWarning($"[Contracts] Failed to parse AnimationRequest JSON: {ex.Message}");
                return new AnimationRequest();
            }
        }

        public string ToJson() => JsonUtility.ToJson(this);
    }
}
